webxr/openxr/input.rs

```diff
@@ -512,22 +554,47 @@ impl OpenXRInput {
 
         let input_changed = buttons_changed || axes_changed;
 
-        let (click_is_active, click_event) =
+        let (click_is_active, mut click_event) = if !use_alternate_input_source {
             self.click_state
-                .update(&self.action_click, session, menu_selected);
+                .update_from_action(&self.action_click, session, menu_selected)
+        } else {
+            (true, None)
+        };
         let (squeeze_is_active, squeeze_event) =
             self.squeeze_state
-                .update(&self.action_squeeze, session, menu_selected);
+                .update_from_action(&self.action_squeeze, session, menu_selected);
+
+        let mut aim_state: Option<HandTrackingAimStateFB> = None;
+        let hand = self.hand_tracker.as_ref().and_then(|tracker| {
+            Some(locate_hand(
```

Comment:
and_then + Some is the same is equivalent to map, right?

Reply:
There's a subtle difference (after removing unneeded ? and Some()), using map+locate_hand returns an `Option<Option<Box<Hand<JointFrame>>>>`, using and_then+locate_hand flattens it to just `Option<Box<Hand<JointFrame>>>`